Added arbitrary plugin configuration through default attributes.

jameskyle opened the pull request.

jameskyle committed:
Updated the collectd cookbook for arbitrary plugin creation using a
node's default attributes. Also added a "filters" conf file.
Finally,bumped the version number.

jameskyle committed:
updated the readme to account for new functionality

coderanger commented on templates/default/filters.conf.erb:
```diff
@@ -0,0 +1,32 @@
+# cat filters.conf
```
1) did you mean to include this line. 2) Why is this file here?

jameskyle committed:
Refactord cookbook. Plugins are now built in plugins recipe for both
servers and clients. Server/Client role is now dtermined by the network
plugin configuration. Version bumped to reflect changes.

jameskyle commented:
Can't believe it's been open this long, hehe. This PR is older than my kids.
